### ISSUE TITLE
Changing your search should't scroll to the first result

### ISSUE THREAD

**Issue by HenryMarshall** [hbaughman/deep-search#7]
## Steps to reproduce
1. Perform a shallow search for something with multiple results on the page
2. Click "Find" until the page scrolls in order to display it
3. Delete the last character of your search (thus recalculating all matches)

### Expected result
A substring of the string that *was* selected after step 2 should now be selected. This is how the baked-in chrome search functions

### Observed result
The first match for the new string will be matched. This will scroll your viewport to see it.

## Notes

Currently the `deepSearch-current-highlight` class is assigned as part of the call to `findAndReplaceDomElements`.  Because resolving #5 makes it impossible to know what will be in the viewport during this replacement, identification of the current highlight must be deferred.
